data-prepper-plugins/otel-trace-raw-processor/src/test/java/com/amazon/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessorTest.java

```diff
@@ -1,40 +1,120 @@
 package com.amazon.dataprepper.plugins.processor.oteltrace;
 
+import com.amazon.dataprepper.metrics.MetricNames;
 import com.amazon.dataprepper.model.configuration.PluginSetting;
 import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.metrics.MetricsTestUtil;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.protobuf.util.JsonFormat;
+import io.micrometer.core.instrument.Measurement;
 import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span;
 import org.assertj.core.api.Assertions;
-import org.junit.jupiter.api.Test;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.Before;
+
+import static org.mockito.Mockito.mock;
 
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.StringJoiner;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
 
 public class OTelTraceRawProcessorTest {
 
     private final static ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-    private final OTelTraceRawProcessor oTelTraceRawProcessor = new OTelTraceRawProcessor(new PluginSetting(null, Collections.EMPTY_MAP));
+    PluginSetting pluginSetting;
+    public OTelTraceRawProcessor oTelTraceRawProcessor;
+
+    @Before
+    public void setup() {
+        pluginSetting = new PluginSetting("OTelTrace", Collections.EMPTY_MAP);
+        pluginSetting.setPipelineName("pipelineOTelTrace");
+        oTelTraceRawProcessor = new OTelTraceRawProcessor(pluginSetting);
+    }
+
+    @Test
+    public void testResourceSpansProcessingErrorMetrics() {
+
+        MetricsTestUtil.initMetrics();
+
+        Record<ExportTraceServiceRequest> mockRecord = mock(Record.class);
+        ExportTraceServiceRequest mockData = mock(ExportTraceServiceRequest.class);
+        ResourceSpans mockResourceSpans = mock(ResourceSpans.class);
+        List<ResourceSpans> mockResourceSpansList = Collections.singletonList(mockResourceSpans);
+
+        when(mockRecord.getData()).thenReturn(mockData);
```

Comment:
Nit: There is some unnecessary mocking happening here. The only class that needs to be mocked is the ResourceSpans class

Reply:
- So the reason I have mocked the Record<ExportTraceServiceRequest> is because I need to pass it as an argument when calling the doExecute() function. 
- The ExportTraceServiceRequest object is mocked separately since the for loop in doExecute() fucntion when traversing through the ResourceSpans has chained method calls and acoording to Mockito documentation, the other cleaner alternative Deep Stubs, should be avoided. Hence I went with mocking individual components.